conf/base.config

```diff
@@ -11,29 +11,95 @@
-  errorStrategy = { task.exitStatus in [143,137] ? 'retry' : 'finish' }
+  errorStrategy = { task.exitStatus in [143,137,104,134,139] ? 'retry' : 'finish' }
   maxRetries = 1
   maxErrors = '-1'
 
   // Process-specific resource requirements
-  $fastqc {
-    errorStrategy = { task.exitStatus in [143,137] ? 'retry' : 'ignore' }
+  withName: busco_download_db {
+    time = 4.h
   }
-  $multiqc {
-    executor = 'local'
+  withName: busco {
     errorStrategy = { task.exitStatus in [143,137] ? 'retry' : 'ignore' }
   }
+  withName: phix_download_db {
+    time = 4.h
+  }
+  withName: porechop {
+    cpus = { check_max (4 * task.attempt, 'cpus' ) }
+    memory = { check_max (30.GB * task.attempt, 'memory' ) }
+    time = { check_max (4.h * task.attempt, 'time' ) }
+  }
+  withName: nanolyse {
+    cpus = { check_max (2 * task.attempt, 'cpus' ) }
+    memory = { check_max (10.GB * task.attempt, 'memory' ) }
+    time = { check_max (3.h * task.attempt, 'time' ) }
+  }
+  //filtlong: exponential increase of memory and time with attempts
+  withName: filtlong {
+    cpus = { check_max (8 * task.attempt, 'cpus' ) }
+    memory = { check_max (64.GB * (2**(task.attempt-1)), 'memory' ) }
+    time = { check_max (24.h * (2**(task.attempt-1)), 'time' ) }
+  }
+  withName: remove_phix {
+    cpus = { check_max (4 * task.attempt, 'cpus' ) }
+    memory = { check_max (8.GB * task.attempt, 'memory' ) }
+    time = { check_max (6.h * task.attempt, 'time' ) }
+  }
+  withName: centrifuge {
+    cpus = { check_max (8 * task.attempt, 'cpus' ) }
+    memory = { check_max (40.GB * task.attempt, 'memory' ) }
+    time = { check_max (12.h * task.attempt, 'time' ) }
+  }
+  withName: kraken2 {
+    cpus = { check_max (8 * task.attempt, 'cpus' ) }
+    memory = { check_max (40.GB * task.attempt, 'memory' ) }
+    time = { check_max (12.h * task.attempt, 'time' ) }
+  }
+  withName: krona {
+    cpus = { check_max (8 * task.attempt, 'cpus' ) }
+    memory = { check_max (20.GB * task.attempt, 'memory' ) }
+    time = { check_max (12.h * task.attempt, 'time' ) }
+  }
+  withName: cat {
+    cpus = { check_max (8 * task.attempt, 'cpus' ) }
+    memory = { check_max (40.GB * task.attempt, 'memory' ) }
+    time = { check_max (12.h * task.attempt, 'time' ) }
+  }
+  withName: megahit {
+    cpus = { check_max (8 * task.attempt, 'cpus' ) }
+    memory = { check_max (40.GB * task.attempt, 'memory' ) }
+    time = { check_max (16.h * task.attempt, 'time' ) }
+  }
+  //SPAdes returns error(1) if it runs out of memory (and for other reasons as well...)!
+  //exponential increase of memory and time with attempts
+  withName: spades {
+    cpus = { check_max (10 * task.attempt, 'cpus' ) }
+    memory = { check_max (64.GB * (2**(task.attempt-1)), 'memory' ) }
+    time = { check_max (24.h * (2**(task.attempt-1)), 'time' ) }
+    errorStrategy = { task.exitStatus in [143,137,1] ? 'retry' : 'ignore' }
+  }
+  withName: spadeshybrid {
+    cpus = { check_max (10 * task.attempt, 'cpus' ) }
```

Comment:
You could probably make this config quite a bit shorter by using labels instead of referring to each process by name. But this should be fine as it is for now 👍 (next hackathon..??)